src/xdg.c

```diff
@@ -400,6 +400,12 @@ xdg_toplevel_view_set_fullscreen(struct view *view, bool fullscreen)
 		fullscreen);
 }
 
+static void
+xdg_toplevel_view_set_tiled(struct view *view, bool tiled)
+{
+	wlr_xdg_toplevel_set_tiled(xdg_toplevel_from_view(view), tiled);
```

Comment:
`subprojects/wlroots/include/wlr/types/wlr_xdg_shell.h`:
```c
/**
 * Request that this toplevel consider itself in a tiled layout and some
 * edges are adjacent to another part of the tiling grid. `tiled_edges` is a
 * bitfield of enum wlr_edges. Returns the associated configure serial.
 */
uint32_t wlr_xdg_toplevel_set_tiled(struct wlr_xdg_toplevel *toplevel,
                uint32_t tiled_edges);
```

So we'd need to convert `view->tiled` to that bitset.
Another question is how / if we should handle `view->tiled_region` here.

Reply:
https://github.com/swaywm/sway/blob/e8c421e917ac59cdf50e5fd366155ea875e4a29e/sway/desktop/xdg_shell.c#L181-L185

Reply:
Especially with `SnapToEdge` I think we could be a bit smarter here (e.g. only set the 3 edges we are actually snapped to).

Reply:
Welp, I completely missed that this was not a Boolean.

I restructured this to set all edges tiled for region-snapped views, but only set the three snapped edges for edge-snapped views. This seems to do the right thing for foot (which doesn't actually care about the tiling direction.)